src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -1220,10 +1224,6 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			} else {
 				executeTaskResult = await this._executeTask(task, resolver, runSource);
 			}
-			if (runSource === TaskRunSource.User) {
-				const workspaceTasks = await this.getWorkspaceTasks();
-				RunAutomaticTasks.runWithPermission(this, this._storageService, this._notificationService, this._workspaceTrustManagementService, this._openerService, this._configurationService, workspaceTasks);
```

Comment:
I think this used to just ask for permission, not run. The idea being that after the user has run a task (`TaskRunSource.User`) they are in the mode of running tasks so it's a good idea to ask them for permission.